enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -94,9 +95,9 @@ static void populateRemoveAndProtectedHeaders(RequestContext requestContext) {
 
         // Choreo-API-Key is considered as a protected header, hence header value should be treated
         // same as other security headers.
-        if (ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeader() != null) {
-            requestContext.getProtectedHeaders().add(ConfigHolder.getInstance().getConfig().getApiKeyConfig()
-                    .getApiKeyInternalHeader().toLowerCase());
+        if (ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders() != null) {
```

Comment:
IMO, we should only remove whatever the single internal-key from what the customer is used.

Ex: we have choreo-api-key and api-key

User uses choreo-api-key as test-key and send some other header to backend with api-key header. This way both headers get removed. I wonder if that is okay.

Reply:
Even though the config name is referred as the internAPIKey header, this header is being used by external requests as well, since we support API-Key for public APIs as well. Hence I think we should treat both of these headers as Choreo feature specific security headers and drop at GW.

Reply:
Well, let's say your call is having Authorization Header, and gateway validates it. Then it would drop only the Authorization header. If we bring the same analogy here, if choreo-api-key is used, you should not drop api-key.